pkg/handler/v2/routes_v2.go

```diff
@@ -5281,8 +5303,133 @@ func (r Routing) validateEKSCredentials() http.Handler {
 		endpoint.Chain(
 			middleware.TokenVerifier(r.tokenVerifiers, r.userProvider),
 			middleware.UserSaver(r.userProvider),
-		)(provider.EKSValidateCredentialsEndpoint(r.presetProvider, r.userInfoGetter)),
-		provider.DecodeEKSTypesReq,
+		)(externalcluster.EKSValidateCredentialsEndpoint(r.presetProvider, r.userInfoGetter)),
+		externalcluster.DecodeEKSTypesReq,
+		handler.EncodeJSON,
+		r.defaultServerOptions()...,
+	)
+}
+
+// swagger:route GET /api/v2/projects/{project_id}/kubernetes/clusters/{cluster_id}/providers/eks/amitypes eks listEKSAMITypesNoCredentials
+//
+//     Gets the EKS AMI types for node group.
+//
+//
+//     Produces:
+//     - application/json
+//
+//     Responses:
+//       default: errorResponse
+//       200: EKSAMITypes
+//       401: empty
+//       403: empty
+func (r Routing) listEKSAMITypesNoCredentials() http.Handler {
+	return httptransport.NewServer(
+		endpoint.Chain(
+			middleware.TokenVerifier(r.tokenVerifiers, r.userProvider),
+			middleware.UserSaver(r.userProvider),
+		)(externalcluster.EKSAMITypesWithClusterCredentialsEndpoint()),
+		common.DecodeEmptyReq,
```

Comment:
you can use this method `common.DecodeEmptyReq` because you have to decode {project_id} and {cluster_id}. It causes swagger error